datafusion/sql/src/unparser/expr.rs

```diff
@@ -1516,7 +1520,7 @@ impl Unparser<'_> {
                 not_impl_err!("Unsupported DataType: conversion: {data_type:?}")
             }
             DataType::Dictionary(_, _) => {
-                not_impl_err!("Unsupported DataType: conversion: {data_type:?}")
+                unreachable!()
```

Comment:
```suggestion
                not_impl_err!("Unsupported DataType: conversion: {data_type:?}")
```

Reply:
We shouldn't introduce a potential panic here

Reply:
Got it, let me change it back